lambdaclass/curse_of_myrra
# Gh 1181 hide characters

Closes #1181 

## Motivation

Simplify the way the display the usable characters in the client ( MainScreen models, CharacterList and characterInfo)

Now `MainScreenManager` contains a list of strings `enableCharactersName`, which contains the names of the characters you want to be avaible. This is the variable you should modify to remove or add avaible characters.
In the next iteration we should map this with the json from the backend.

<img width="774" alt="Captura de pantalla 2023-11-27 a la(s) 12 20 55" src="https://github.com/lambdaclass/curse_of_myrra/assets/62818001/5d6f0e6f-1177-4ef0-9d18-421770ad9b4b">


## Summary of changes

Backend:

1. Left only muflus as a playable character

Client:

1. Changed `MainScreenManager`, `UIModelManager`, `CharacterInfoManager` and `CharacterListManager`.
2. Add the method to `Utils` class.
3. Rename all Character UI models variants from name + Variant (ex: Muflus Variant) to just name (ex: Muflus), i made this to simplify the matching from models names and real character names
4. Rename H4ck model name from `H4ck_v2` to `H4ck`

## How has this been tested?

1. In the MainScreenManager script Add/Remove character names in the `enableCharactersName` list
2. Start the game, check if the characters the Add/Remove are avaible or not in the MainScreen models
5. Check if in the CharacterList scene you can click the characters that are in the list and check if you cannot click the characters that are not part of the list
6. Check if in the CharacterInfo screen you  only see the avaibles characters, use the arrows to see them all
7. Finally start a game and you only have to be able to play with muflus, since is the only avaible in the backend

## Test additions / changes

List tests added/updated.

## Checklist
